e2e/isolated/isolated_test.go

```diff
@@ -0,0 +1,247 @@
+				Name:    svcName,
+				EnvName: envName,
+			})
+		})
+
+		It("svc deploy should succeed", func() {
+			Expect(privateEnvDeployErr).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("when running svc show to retrieve the service configuration, resources, and endpoint, then querying the service", func() {
+		var (
+			svc          *client.SvcShowOutput
+			svcShowError error
+		)
+		BeforeAll(func() {
+			svc, svcShowError = cli.SvcShow(&client.SvcShowRequest{
+				Name:      svcName,
+				AppName:   appName,
+				Resources: true,
+			})
+		})
+
+		It("should not return an error", func() {
+			Expect(svcShowError).NotTo(HaveOccurred())
+		})
+
+		It("should return correct configuration", func() {
+			Expect(svc.SvcName).To(Equal(svcName))
+			Expect(svc.AppName).To(Equal(appName))
+			Expect(len(svc.Configs)).To(Equal(1))
+			Expect(svc.Configs[0].Environment).To(Equal(envName))
+			Expect(svc.Configs[0].CPU).To(Equal("256"))
+			Expect(svc.Configs[0].Memory).To(Equal("512"))
+			Expect(svc.Configs[0].Port).To(Equal("80"))
+		})
+
+		Context("when running `svc logs`", func() {
+			It("logs should be displayed", func() {
+				var svcLogs []client.SvcLogsOutput
+				var svcLogsErr error
+				Eventually(func() ([]client.SvcLogsOutput, error) {
+					svcLogs, svcLogsErr = cli.SvcLogs(&client.SvcLogsRequest{
+						AppName: appName,
+						Name:    svcName,
+						EnvName: envName,
+						Since:   "1h",
+					})
+					return svcLogs, svcLogsErr
+				}, "60s", "10s").ShouldNot(BeEmpty())
+
+				for _, logLine := range svcLogs {
+					Expect(logLine.Message).NotTo(Equal(""))
+					Expect(logLine.LogStreamName).NotTo(Equal(""))
+					Expect(logLine.Timestamp).NotTo(Equal(0))
+					Expect(logLine.IngestionTime).NotTo(Equal(0))
+				}
+			})
+		})
+
+		Context("when running `env show --resources`", func() {
+			var envShowOutput *client.EnvShowOutput
+			var envShowErr error
+			It("show show internal ALB", func() {
+				envShowOutput, envShowErr = cli.EnvShow(&client.EnvShowRequest{
+					AppName: appName,
+					EnvName: envName,
+				})
+			})
+			It("should not return an error", func() {
+				Expect(envShowErr).NotTo(HaveOccurred())
+			})
+			It("should now have an internal ALB", func() {
+				Expect(envShowOutput.Resources).To(ContainElement(HaveKeyWithValue("type", "AWS::ElasticLoadBalancingV2::LoadBalancer")))
+			})
+		})
+
+		Context("when `curl`ing the LB DNS", func() {
+			It("is not reachable", func() {
+				cmd := osExec.Command("curl", fmt.Sprintf("http://%s.%s.%s.internal", svcName, envName, appName))
+				cmd.Stdout = os.Stdout
+				cmd.Stderr = os.Stderr
+				Expect(cmd.Stderr).NotTo(BeNil())
+			})
+		})
+
+		Context("when `curl`ing the LB DNS from within the container", func() {
+			It("session manager should be installed", func() {
+				// Use custom SSM plugin as the public version is not compatible to Alpine Linux.
+				err := client.BashExec("chmod +x ./session-manager-plugin")
```

Comment:
How does it work if we don't have `session-manager-plugin` in this suite?

Reply:
Also instead of using exec, can we just create another frontend service and then ping the ALB endpoint in the code? So when we ping the frontend service, it pings the internal ALB to see if it gets ok. That way we also test if internal ALB can work well with public ALB.

Reply:
I like that this test suite imports a VPC with no IGW so we get to have at least a suite where we can add logic to make sure copilot behaves nicely in isolated environments 😍 
Up to y'all, but I would prefer keeping it as is.